pkg/release/caching_getter.go

```diff
@@ -0,0 +1,52 @@
+package release
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type cachingGetter struct {
+	cache.Cache
+}
+
+func NewCachingGetter(cfg *rest.Config) (Getter, error) {
+	scheme := runtime.NewScheme()
+	err := releasev1alpha1.AddToScheme(scheme)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	opts := cache.Options{
+		Scheme: scheme,
+	}
+
+	c, err := cache.New(cfg, opts)
```

Comment:
After I wrote the watch-based client, I had to re-check few implementation details and then I found this, which is exactly what I was working on, but more general. I ditched my original implementation and used this.